lib/ansible/modules/cloud/ovirt/ovirt_vm.py

```diff
@@ -1159,6 +1188,19 @@ def build_entity(self):
             initialization=self.get_initialization() if self.param('cloud_init_persist') else None,
         )
 
+    def _get_export_domain_service(self):
+        provider_name = self._module.params['export_domain']
+        export_sds_service = self._connection.system_service().storage_domains_service()
+        export_sd = search_by_name(export_sds_service, provider_name)
```

Comment:
You can use following, `get_id_by_name` will raise error for you in case not found:
```suggestion
        export_sd_id = get_id_by_name(export_sds_service, provider_name)
        return export_sds_service.service(export_sd_id)
```